src/pykeen/nn/emb.py

```diff
@@ -1072,6 +1072,27 @@ def forward(
         return x
 
 
+class ConcatMLP(nn.Sequential):
+    """An MLP applied to the concatenation of token representations."""
+
+    def __init__(
+        self,
+        num_tokens: int,
+        embedding_dim: int,
+        dropout: float = 0.1,
+    ):
+        super().__init__(
+            nn.Linear(num_tokens * embedding_dim, 2 * embedding_dim),
+            nn.Dropout(dropout),
+            nn.ReLU(),
+            nn.Linear(2 * embedding_dim, embedding_dim),
+        )
+
+    def forward(xs: torch.FloatTensor, dim: int) -> torch.FloatTensor:  # noqa: D102
+        assert dim == -2
```

Comment:
can you add a more substantial error message that says why it has to be -2

Reply:
I moved it to the model file; this is only a private class, which is not thought to be visible from the outside, and only serves as a convenience method to get the configuration from the paper.